numpy/ma/tests/test_core.py

```diff
@@ -685,6 +685,36 @@ def test_mvoid_print(self):
         finally:
             masked_print_option.set_display(ini_display)
 
+    def test_mvoid_multidim_print(self):
+
+        # regression test for gh-6019
+        t_ma = masked_array(data = [([1, 2, 3],)],
+                            mask = [([False, True, False],)],
+                            fill_value = ([999999, 999999, 999999],),
+                            dtype = [('a', '<i8', (3,))])
+        assert str(t_ma[0]) == "([1, --, 3],)"
+        assert repr(t_ma[0]) == "([1, --, 3],)"
```

Comment:
While this PR does fix the str and repr of this array, the array still doesn't seem very useable. 

    >>> 1/t_ma
    TypeError: ufunc 'divide' not supported for the input types, ....

But that can be left for another time.